### DISCUSSION TITLE
How to connect a USB Piano to ESP32c3

### DISCUSSION THREAD

**Discussion by ssk-dev** [lathoub/Arduino-BLE-MIDI#83] [Q&A]
Hi, i am currently using the sketch MidiBle_Client.
My goal is to connect my Seaboard Rise 49 (USB) to my Roland FP60X (BLE).
The communication between the ESP32c3 (supermini china stuff) an the Roland works as expected.
Now the big question -> how to connect the Seaboard with the ESP over USB?

Another idea is to use a ESP as bridge -> a complete BLE connection between Roland and Seaboard to work bidirectional. Need support for the magic code to make it possible.

**Comment by alf45tar** [comment]
May I suggest to look into my [PedalinoMini](https://github.com/alf45tar/PedalinoMini)?
It can do MID bridge between USB/Legacy MIDI/BLE/AppleMIDI/ipMIDI without any new coding.

**Reply by ssk-dev** [reply]
Hi @alf45tar, 
your project is pretty cool and i think i gonna take it later for another project. The ESP32C3 supermini isnt supported
![image](https://github.com/lathoub/Arduino-BLE-MIDI/assets/24464850/16d55575-a7b5-42bb-971a-41b728790ffb)
I found a solution to solve my current project.
That is my setup.
![image](https://github.com/lathoub/Arduino-BLE-MIDI/assets/24464850/f77d6b71-1fee-4f3a-a757-ab3d0cdec979)
Sender:
- Connect to Sender -> BLEMIDI_CREATE_INSTANCE("f2:c1:d9:36:e7:6b",MIDI)
- Add listener to read incoming midi messages form BLE like e.g. -> MIDI.setHandleNoteOn()
- Send messages to HardwareSerial (RX = GPPIO20, TX GPIO21)

Receiver:
- Connect to Sender -> BLEMIDI_CREATE_INSTANCE("7b:c1:d9:36:e7:6b",MIDI)
- Add function in the loop to read incoming midi messages from HardwareSerial like e.g. -> HS_MIDI.handleIncomingMessage()
- Map the Incomming messages by type byte e.g. 128 = NoteOn and send Midi.sendNoteOn(128, 60, 1) to receiving BLE Device

Next steps will be to add on sendersside GPIO'S some fader and buttons to manipulate the Midievents.

**Comment by lukewakeford** [comment]
@ssk-dev If you are able to share your sketch or some of it, it could help me out as I'm currently struggling to get anything to connect to an ESP32-c3 SuperMini -> https://github.com/lathoub/Arduino-BLE-MIDI/discussions/89
